lib/server.js

```diff
@@ -53,12 +53,22 @@ const start = async function start() {
   }
 
   const { app } = this;
-  const { host, middleware, port } = this.options;
+  const { host, middleware, port, waitForBuild } = this.options;
   const builtins = getBuiltins(app, this.options);
 
   this.options.host = await host;
   this.options.port = await port;
 
+  if (waitForBuild) {
+    app.use(async (ctx, next) => {
+      if (!this.compiled) {
```

Comment:
since the next line is just awaiting a promise, we don't need a logical check here. we can always await the promise. once the promise is resolved, this will pass-through. I'll push this change to your branch.